docs/contributing/docs-and-website-components.md

```diff
@@ -0,0 +1,418 @@
+
+The `<TableOfContents />` component is used to render a list of subheaders from a docs page and automatically provide deep links to them.
+
+#### Usage
+
+The component takes 2 props:
+
+- `location` - an object provided in the props of page templates by default
+- `page` - an object with data passed in from the sites `gatsby-node.js` that contains information from the MDX plugin about the structure of headings
+
+<!-- prettier-ignore -->
+```javascript
+import TableOfContents from "../../www/src/components/docs-table-of-contents"
+
+<TableOfContents location={props.location} page={page} />
+```
+
+_You can also refer to [an example of usage of the Table of Contents in the Gatsbyjs.org source code](https://github.com/gatsbyjs/gatsby/blob/1d65ce051967dda5c4a89da920fc34692524e237/www/src/templates/template-docs-markdown.js#L121)_
+
+#### Sample
+
+The Table of Contents looks like this when rendered (and is also displayed on the right hand side of the page):
+
+<TableOfContents
+location={{ pathname: "/contributing/docs-and-website-components/" }}
+page={{
+    frontmatter: {
+      title: "Docs and Website Components",
+      tableOfContentsDepth: 2,
+    },
+    tableOfContents: {
+      items: [
+        {
+          url: "#globally-available-components",
+          title: "Globally available components",
+          items: [
+            {
+              url: "#guide-list",
+              title: "Guide list",
+            },
+            {
+              url: "#egghead-embed",
+              title: "Egghead embed",
+            },
+            {
+              url: "#pull-quote",
+              title: "Pull quote",
+            },
+          ],
+        },
+        {
+          url: "#other-available-components",
+          title: "Other available components",
+          items: [
+            {
+              url: "#layer-model",
+              title: "Layer model",
+            },
+            {
+              url: "#horizontal-navigation-list",
+              title: "Horizontal navigation list",
+            },
+            {
+              url: "#breadcrumb",
+              title: "Breadcrumb",
+            },
+            {
+              url: "#table-of-contents",
+              title: "Table of Contents",
+            },
+          ],
+        },
+        {
+          url: "#writing-content-in-markdown",
+          title: "Writing content in markdown",
+          items: [
+            {
+              url: "#code-blocks",
+              title: "Code blocks",
+            },
+          ],
+        },
+        {
+          url: "#styling-components-on-gatsbyjsorg-with-theme-ui",
+          title: "Styling components on Gatsbyjs.org with Theme-UI",
+          items: [
+            {
+              url: "#design-tokens",
+              title: "Design tokens",
+            },
+            {
+              url: "#the-sx-prop",
+              title: "The sx prop",
+            },
+          ],
+        },
+      ],
+    },
+  }}
+/>
+
```

Comment:
Here's something I ran into today, how do you turn off the Table of Contents widget in a given Markdown doc? That seems more likely of a scenario than working with it directly.